connections/localproxy/https-proxy.go

```diff
@@ -138,7 +140,11 @@ func (thisRef *HTTPSProxy) Disconnect() {
 		Details: fmt.Sprintf("%s, %s", thisRef.tcpType, thisRef.source),
 	})
 
-	thisRef.localListener.Close()
+	thisRef.currentEventCotextSync.Lock()
```

Comment:
```suggestion
	thisRef.currentEventContextSync.Lock()
```

`cotext` is a feminine hygene product 😉